### ISSUE TITLE
VMSS creation fails since v2.97.0 when using a data_disk block

### ISSUE THREAD

**Issue by fabiendelpierre** [hashicorp/terraform-provider-azurerm#15516]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
% terraform -version
Terraform v1.1.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.97.0
```

### Affected Resource(s)

* `azurerm_linux_virtual_machine_scale_set`

I'd suggest to whomever looks into this to check the resource `azurerm_windows_virtual_machine_scale_set` as well, I didn't take the time to test, but it stands to reason it would be affected in the same way.

### Terraform Configuration Files

```hcl
resource "azurerm_linux_virtual_machine_scale_set" "example" {
  name                = "example-vmss"
  resource_group_name = data.azurerm_resource_group.my_rg.name
  location            = data.azurerm_resource_group.my_rg.location
  sku                 = "Standard_B2ms"
  instances           = 1
  admin_username      = "adminuser"

  admin_ssh_key {
    username   = "adminuser"
    public_key = file("~/.ssh/id_rsa.pub")
  }

  source_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04-LTS"
    version   = "latest"
  }

  os_disk {
    storage_account_type = "StandardSSD_LRS"
    caching              = "ReadWrite"
  }

  data_disk {
    storage_account_type  = "StandardSSD_LRS"
    caching               = "ReadOnly"
    disk_size_gb          = 32
    lun                   = "10"
  }

  network_interface {
    name    = "example"
    primary = true

    ip_configuration {
      name      = "internal"
      primary   = true
      subnet_id = azurerm_subnet.example.id
    }
  }
}
```

### Debug Output

Sorry, I didn't do this because the debug output spits out a bunch of data about my environment and I don't feel like going through it and scrubbing it. I think this shouldn't be a problem given the simplicity of the issue.

### Expected Behaviour

The VMSS should be created.

### Actual Behaviour

The following error is returned:

│ Error: updating Linux Virtual Machine Scale Set "example-vmss" (Resource Group "my-rg"): compute.VirtualMachineScaleSetsClient#Update: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="Property 'dataDisk.diskIOPSReadWrite' can be set only when property 'additionalCapabilities.ultraSSDEnabled' is enabled." Target="dataDisk.diskIOPSReadWrite"

FWIW this was reported to me by a coworker, for whom it presented with this error: 

Error: setting `data_disk`: Invalid address to set: []string{"data_disk", "0", "ultra_ssd_mbps_read_write"}

I wasn't able to reproduce the issue and get the same error as my coworker, but I believe the two stem from the same problem.

It's like it's defaulting to setting a value for the `disk_iops_read_write` or `disk_mbps_read_write` settings, which can only be used when creating an Ultra disk, but I'm not working with Ultra disks.

### Steps to Reproduce

Just set up a resource group, a vnet and a subnet, then use the example above to create a VM scale set in the subnet.
With v2.96.0 of the AzureRM provider, it'll work.
With v2.97.0, it'll fail with the above error. And if you remove the `data_disk` block, it'll work.

### Important Factoids

Probably nothing. This is a bog standard Azure public cloud (not Gov) subscription. Using region East US 2 but I don't think that's relevant.

**Comment by Divya1388** [comment]
Facing the same issue when using premium_LRS data disks

**Comment by WodansSon** [comment]
@fabiendelpierre Thank you for opening this issue. I took a look at was appears to be going on is that in preparation of the property being renamed to `ultra_ssd_disk_iops_read_write` in v3.0 the fields were marked as optional computed. Fields act very differently when they are marked as optional computed and can be a bit tricky to deal with. So what was happening is that they were being initialized as `0`, since they were no longer `nil` they would be persisted and get passed to the RP during the create call when they previously were not. Anyway, I now check the fields to make sure they are greater than `0` before they will be included in the payload we send to the RP.

**Comment by WodansSon** [comment]
@Divya1388 I also verified my fix against a `Premium_LRS` data disk and it fixes that issue as well. 🙂

**Comment by darrens280** [comment]
Similar to the original post, when deploying a Scale Set (azurerm_windows_virtual_machine_scale_set) using a custom image which has a data disk already included. The Terraform documentation states that a data_disk block is required in the scale set resource, this causes conflict during APPLY, since the source_image_id should negate the need for a data_disk block (v2.97.0).

I can confirm this is now resolved when testing v2.98.0 of AzureRM Provider

Thank you
